Assets/Src/Scripts/Animator/PoseLandmarksRecognizer.cs

```diff
@@ -24,47 +26,81 @@ public class PoseLandmarksRecognizer {
-      Vector3 rightElbow = toVector(poseLandmarks.Landmark[Landmarks.RightElbow]);
-      Vector3 rightWrist = toVector(poseLandmarks.Landmark[Landmarks.RightWrist]);
+      PoseLandmarks landmarks = new PoseLandmarks();
 
-      Vector3 forward = GetNormal(leftShoulder, leftHip, rightHip);
+      if (isBodyVisibe(poseLandmarks)) {
+        Vector3 leftHip =
+            _filterDict["leftHip"].Update(toVector(poseLandmarks.Landmark[Landmarks.LeftHip]));
+        Vector3 rightHip =
+            _filterDict["rightHip"].Update(toVector(poseLandmarks.Landmark[Landmarks.RightHip]));
+        Vector3 leftShoulder = _filterDict["leftShoulder"].Update(
+            toVector(poseLandmarks.Landmark[Landmarks.LeftShoulder]));
+        Vector3 leftElbow =
+            _filterDict["leftElbow"].Update(toVector(poseLandmarks.Landmark[Landmarks.LeftElbow]));
+        Vector3 leftWrist =
+            _filterDict["leftWrist"].Update(toVector(poseLandmarks.Landmark[Landmarks.LeftWrist]));
+        Vector3 rightShoulder = _filterDict["rightShoulder"].Update(
+            toVector(poseLandmarks.Landmark[Landmarks.RightShoulder]));
+        Vector3 rightElbow = _filterDict["rightElbow"].Update(
+            toVector(poseLandmarks.Landmark[Landmarks.RightElbow]));
+        Vector3 rightWrist = _filterDict["rightWrist"].Update(
+            toVector(poseLandmarks.Landmark[Landmarks.RightWrist]));
 
-      // Note: left and right are mirrored here.
-      PoseLandmarks landmarks = new PoseLandmarks();
-      landmarks.Add(
-          new PoseLandmark { Id = Landmarks.Hip, Rotation = Quaternion.Euler(0, 180, 0) });
-      landmarks.Add(new PoseLandmark { Id = Landmarks.LeftShoulder,
-                                       Rotation = Quaternion.LookRotation(
-                                           rightShoulder - rightElbow, -forward) });
-      landmarks.Add(new PoseLandmark { Id = Landmarks.LeftElbow,
-                                       Rotation = Quaternion.LookRotation(
-                                           rightElbow - rightWrist, rightShoulder - rightElbow) });
-      landmarks.Add(new PoseLandmark { Id = Landmarks.RightShoulder,
-                                       Rotation = Quaternion.LookRotation(leftShoulder - leftElbow,
-                                                                          -forward) });
-      landmarks.Add(new PoseLandmark { Id = Landmarks.RightElbow,
-                                       Rotation = Quaternion.LookRotation(
-                                           leftElbow - leftWrist, leftShoulder - leftElbow) });
+        Vector3 forward = GetNormal(leftShoulder, leftHip, rightHip);
+
+        // Note: left and right are mirrored here.
+        if (Vector3.Angle(forward, Vector3.forward) < 45.0f) {
+          landmarks.Add(
+              new PoseLandmark { Id = Landmarks.Hip, Rotation = Quaternion.LookRotation(forward) });
+        }
+
+        landmarks.Add(new PoseLandmark { Id = Landmarks.LeftShoulder,
```

Comment:
The four code pieces (lines 75-77, 78-81, 82-84, 85-88) are not using the same indention pattern.

I think you can format all of them as the second one:

```
        landmarks.Add(new PoseLandmark {
          Id = Landmarks.LeftElbow,
          Rotation = Quaternion.LookRotation(rightElbow - rightWrist, rightShoulder - rightElbow)
        });
```

Reply:
clang-format Linter don‘t recommend like this， It throw the error， I think clang-format is also not completed